eli5/sklearn/transform.py

```diff
@@ -0,0 +1,179 @@
+def _pipeline_names(est, in_names=None):
+    names = in_names
+    for name, trans in est.steps:
+        if trans is not None:
+            names = transform_feature_names(trans, names)
+    return names
+
+
+@transform_feature_names.register(FeatureUnion)
+def _union_names(est, in_names=None):
+    return ['{}:{}'.format(trans_name, feat_name)
+            for trans_name, trans, _ in est._iter()
+            for feat_name in transform_feature_names(trans, in_names)]
+
+
+@transform_feature_names.register(SelectorMixin)
+def _select_names(est, in_names=None):
+    mask = est.get_support(indices=False)
+    in_names = _get_feature_names(est, feature_names=in_names,
+                                  num_features=len(mask))
+    return [in_names[i] for i in np.flatnonzero(mask)]
+
+
+# XXX Should these generic things be in eli5.transform?
+# I've left them here to make use of eli5.sklearn.utils.get_feature_names
+
+
+def _formatted_names(fmt, get_n_inputs=None):
+    def transform_names(est, in_names=None):
+        if get_n_inputs:
+            in_names = _get_feature_names(est, feature_names=in_names,
+                                          num_features=get_n_inputs(est))
+        print(in_names[:100])
+        return [fmt.format(name) for name in in_names]
+    return transform_names
+
+
+def _component_names(fmt, attr):
+    def transform_names(est, in_names=None):
+        return [fmt.format(i) for i in range(getattr(est, attr))]
+    return transform_names
+
+
+def make_tfn_featurewise(func_name):
+    return _formatted_names(func_name + '({})')
+
+
+def make_tfn_components(func_name, n_components_attr):
+    return _component_names(func_name + '{}', n_components_attr)
+
+
+def make_tfn_weighted(weight_attr=None, max_features=3,
+                      threshold=None, abs=False,
+                      show_weights=True,
+                      func_name=None):
+
+    if max_features and weight_attr is None:
+        raise ValueError('Require weight_attr if max_features != 0')
+
+    if threshold is None:
+        threshold = 0. if abs else -np.inf
+
+    if show_weights:
+        if show_weights is True:
+            show_weights = '0.2g'
+        feat_fmt = '{weight:%s}*{name}' % show_weights
+    else:
+        feat_fmt = '{name}'
+    feat_join = '+'
+
+    if max_features and func_name is not None:
+        fmt = '{func_name}{i}=({feats})'
+    elif max_features:
+        fmt = '({feats})'
+    else:
+        fmt = '{func_name}{i}'
+
+    def transform_names(est, in_names=None):
+        weights = orig_weights = getattr(est, weight_attr)
+        in_names = _get_feature_names(est, feature_names=in_names,
+                                      num_features=weights.shape[1])
+        if max_features:
+            # TODO: support sparse weights
+            if abs:
+                weights = np.abs(weights)
+            order = np.argsort(weights, axis=1)[::-1]
+            top_names = np.take(in_names, order[:, :max_features])
```

Comment:
probably not too important, but it seems this could be made faster using [eli5.utils.argsort_k_largest](https://github.com/TeamHG-Memex/eli5/blob/e2f2acfeac5f7a13f13b314e486069660287d7b0/eli5/utils.py#L6)